### PR TITLE
Fixing squid: :SwitchLastCaseIsDefaultCheck switch statements should end with a default clause part 1

### PR DESCRIPTION
**Pull request by devFozgul**
This pull request is focused on resolving occurrences of Sonar rule 
SwitchLastCaseIsDefaultCheck - "switch statements should end with a default clause "
 You can find more information about the issue here: 
 https://dev.eclipse.org/sonar/rules/show/squid:SwitchLastCaseIsDefaultCheck
 Please let me know if you have any questions.
Fevzi Ozgul

### CODE REVIEW

**Comment by elifoster** [comment]
Most Of this pr isn't even related to switch statements...

**Comment by devFozgul** [comment]
Hi,
Why do you think so?  Every fixing I have done in this PR is  within the
switch block.
Regards,

**Comment by elifoster** [comment]
Have you looked at the diff?...

**Comment by devFozgul** [comment]
Hi Eli, Yes I did.  My fix adds a default clause within the switch block.
Could you please be more specific about what I am not seeing. Thank you.

**Comment by elifoster** [comment]
1. Diamond operators got updated to the java 7 syntax even though you already opened a PR for that

2. There are like ten *.rej files.

3. Conditional statements style got updated (replacing true/false with just the member or !member).

**Comment by devFozgul** [comment]
I really need to check that. It seems we are not talking about the same
thing. Thank you for the feedback.

File: src/main/java/flaxbeard/steamcraft/client/render/RenderCanister.java [hunk 1]
```diff
@@ -35,10 +35,10 @@ public void doRender(Entity var1, double x, double y, double z, float var8, floa
         item.hoverStart = 0.0F;
 
         GL11.glTranslated(0, 0.85F - (4F / 16F), 0);
-        if (this.renderManager.options.fancyGraphics == false) {
+        if (!this.renderManager.options.fancyGraphics) {
             GL11.glScalef(1.25F, 1.25F, 1.25F);
         }
-        if (this.renderManager.options.fancyGraphics == true || item.getEntityItem().getItem() instanceof ItemBlock) {
```
Comment (xbony2) [inline src/main/java/flaxbeard/steamcraft/client/render/RenderCanister.java]: thank you

(Santa, if you were the one who put "== true" in I'm gonna yell at you)

**Comment by xbony2** [comment]
@devFozgul remove the .ref files and this will be good to merge ^_^

**Comment by elifoster** [comment]
@xbony2 Not quite. The changes of `<Whatever>` to `<>` need to be removed as well, because there's already a PR for that #432

**Comment by devFozgul** [comment]
Sorry but I do not understand what you mean by .ref files.  I know that your "codechicken/lib" should not be touched.  Please make  it clear.  By the way the PR #441 was only one part of the same problem fixed, amounting to a Technical Dept of 1hr. There is actually 4h TD for this issue in the code.  There are so many more places with the same issue. I sent that   portion  to you first only to get your reaction. I will send the rest with a different PR.

File: src/main/java/flaxbeard/steamcraft/client/render/RenderCanister.java (hunk 1 above)
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/client/render/RenderCanister.java]: I wasn't.

File: src/main/java/flaxbeard/steamcraft/codechicken/lib/raytracer/RayTracer.java.rej [hunk 2]
```diff
@@ -0,0 +1,10 @@
+diff a/src/main/java/flaxbeard/steamcraft/codechicken/lib/raytracer/RayTracer.java b/src/main/java/flaxbeard/steamcraft/codechicken/lib/raytracer/RayTracer.java	(rejected hunks)
+@@ -20,7 +20,7 @@
+ import java.util.List;
+ 
+ public class RayTracer {
+-    private static ThreadLocal<RayTracer> t_inst = new ThreadLocal<RayTracer>();
++    private static ThreadLocal<RayTracer> t_inst = new ThreadLocal<>();
+     private Vector3 vec = new Vector3();
+     private Vector3 vec2 = new Vector3();
+     private Vector3 s_vec = new Vector3();
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/codechicken/lib/raytracer/RayTracer.java.rej]: @devFozgul This is one of the .ref files.

**Comment by devFozgul** [comment]
This is all mixed up. I will close this and create a new PR for SwitchLastCaseIsDefaultCheck  issue.

**Pull request closed without merging**